contracts/contracts/sbtc-deposit.clar

```diff
@@ -46,28 +48,39 @@
         ;; Assert that the deposit has not already been completed (no replay)
         (asserts! (is-none replay-fetch) ERR_DEPOSIT_REPLAY)
 
+        ;; Verify that Bitcoin hasn't forked by comparing the burn hash provided
+        (asserts! (is-eq (some burn-hash) (get-burn-header burn-height)) ERR_INVALID_BURN_HASH)
+
         ;; Mint the sBTC to the recipient
         (try! (contract-call? .sbtc-token protocol-mint amount recipient))
 
         ;; Complete the deposit
-        (ok (contract-call? .sbtc-registry complete-deposit txid vout-index amount recipient))
+        (ok (contract-call? .sbtc-registry complete-deposit txid vout-index amount recipient burn-hash burn-height))
     )
 )
 
+(define-read-only (get-burn-header (height uint))
+    ;; (get-burn-block-info? header-hash height)
+    (get-tenure-info? burnchain-header-hash height)
+)
+
 ;; Accept multiple new deposit requests
 ;; Note that this function can only be called by the current
 ;; bootstrap signer set address - it cannot be called by users directly.
 ;; 
 ;; This function handles the validation & minting of sBTC by handling multiple (up to 1000) deposits at a time, 
 ;; it then calls into the sbtc-registry contract to update the state of the protocol. 
-(define-public (complete-deposits-wrapper (deposits (list 650 {txid: (buff 32), vout-index: uint, amount: uint, recipient: principal})))
-    (let 
-        (
-            (current-signer-data (contract-call? .sbtc-registry get-current-signer-data))
-        )
-
+(define-public (complete-deposits-wrapper 
+        (deposits (list 650 {txid: (buff 32), vout-index: uint, amount: uint, recipient: principal, burn-hash: (buff 32), burn-height: uint}))
+        (burn-height uint)
+        (burn-hash (buff 32))
```

Comment:
I don't think we need the `burn-height` and `burn-hash` again here on line 75-76.